lib/config/fileconf.go

```diff
@@ -302,6 +299,28 @@ func MakeSampleFileConfig(flags SampleFlags) (fc *FileConfig, err error) {
 	return fc, nil
 }
 
+func setJoinParams(g *Global, flags SampleFlags) error {
+	joinMethod := flags.JoinMethod
+	if joinMethod == "" && flags.AuthToken != "" {
+		joinMethod = string(types.JoinMethodToken)
+	}
+	g.JoinParams = JoinParams{
+		TokenName: flags.AuthToken,
+		Method:    types.JoinMethod(joinMethod),
+	}
+	params, err := client.ParseJoinParamsSpec(flags.JoinParams)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for k, v := range params {
```

Comment:
As mentioned above, I would just create a specific flag for the Azure client ID and avoid generic "join params".